File: 10-11-2023_Check_for_Balanced_Tree.java
class Tree {
    // Function to check whether a binary tree is balanced or not.
    public int height(Node root) {
        if(root == null) {
            return 0;
        }
        return 1 + Math.max(height(root.left), height(root.right));
    }

    public boolean isBalanced(Node root) {
        if(root == null) {
            return true;
        }
        int left_height = height(root.left);
        int right_height = height(root.right);
        if((Math.abs(left_height - right_height) == 0 || Math.abs(left_height - right_height) == 1) && isBalanced(root.left) && isBalanced(root.right)) {
            return true;
        }
        return false;
    }
}

File: 10-11-2023_Check_for_Balanced_Tree.cpp
class Solution{
    public:
    //Function to check whether a binary tree is balanced or not.
    int height(Node *root){
        if(root == NULL){
            return 0;
        }
        return 1 + max(height(root -> left), height(root -> right));
    }
    bool isBalanced(Node *root){
        if(root == NULL){
            return true;
        }
        int left_height = height(root -> left);
        int right_height = height(root -> right);
        if((abs(left_height - right_height) == 0 or abs(left_height - right_height) == 1) and isBalanced(root -> left) and isBalanced(root -> right)){
            return true;
        }
        return false;
    }
};
